Use the CMDB v3 /items/system endpoint

rowanmanning opened the pull request:
We've had to switch endpoints for the move from v2 to v3 of CMDB. The intention is to support filtering in the URL at a later date which may mean we can increase the per-page limit and only request the data we need.

rowanmanning committed:
Use the CMDB v3 /items/system endpoint

rowanmanning commented:
@GeoffThorpeFT, not sure if you want to look over this

GeoffThorpeFT commented on lib/fetch-system-codes.js:
```diff
@@ -16,8 +16,10 @@ function fetchSystemCodes(apiKey, page = 1, systemCodes = []) {
+			return response.body
+				.filter(entry => entry.systemCode)
+				.map(entry => entry.systemCode);
 		})
```
I assume the above is aiming to extract just the field named "systemCode" from the rather long list of fields returned by the v3/items/system call. However, due to a legacy issue with data uploads (too long to explain) the systemcode is actually held in the key field "dataItemID" not "systemCode".  You may find some "systemCode" fields but its no longer used.

rowanmanning replied:
Excellent, thanks Geoff! Glad I asked :) will change

GeoffThorpeFT commented on test/unit/lib/fetch-system-codes.js:
```diff
@@ -46,8 +46,9 @@ describe('lib/middleware/fetch-system-codes', () => {
+					{systemCode: 'foo'},
+					{noSystemCode: 'this should not appear'},
+					{systemCode: 'bar'}
 				]
```
Due to above issue with the "systemCode" field I recommend changing these mockups to populate "dataItemID" since that is the real system code field.

rowanmanning committed:
Use the dataItemId property over systemCode

As @GeoffThorpeFT mentioned, systemCode is no longer used. We now get a
more complete list of codes by using dataItemId.
